lab-adam/route/hospital-route.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const Router = require('express').Router;
+const jsonParser = require('body-parser').json();
+const debug = require('debug')('hospital:hospital-route');
+
+const createError = require('http-errors');
+const City = require('../model/city.js');
+const Hospital = require('../model/hospital.js');
+
+const hospitalRouter = module.exports = new Router();
+
+hospitalRouter.get('/api/city/:cityID/hospital/:id', function(req, res, next){
+  debug('running GET route');
+  Hospital.findById(req.params.id)
+  .then(hospital => res.json(hospital))
+  .catch(err => next(createError(404, err.message)));
+});
+
+hospitalRouter.put('/api/city/:cityID/hospital/:id', jsonParser, function(req, res, next){
+  debug('running GET route');
+  Hospital.findByIdAndUpdate(req.params.id, req.body, {new: true})
+  .then(hospital => res.json(hospital))
+  .catch(err => {
+    if (err.name === 'ValidationError') return next(err);
+    next(createError(404, err.message));
+  });
+});
+
+hospitalRouter.post('/api/city/:cityID/hospital', jsonParser, function(req, res, next){
+  debug('running POST route');
+  City.findByIdAndAddHospital(req.params.cityID, req.body)
+  .then(hospital => res.json(hospital))
+  .catch(next);
+});
+
+hospitalRouter.delete('/api/city/:cityID/hospital/:id', function(req, res, next){
+  debug('running DELETE route');
+  Hospital.findById(req.params.id).remove()
+  .then(() => {
+    return City.findById(req.params.cityID);
+  })
+  .then(data => {
```

Comment:
It removes an object from an array in Mongo I believe. Not sure but I think mongoose just inherits this from Mongo?  
https://docs.mongodb.com/manual/reference/operator/update/pull/